src/main/java/seedu/address/model/scheduler/time/HourBlock.java

```diff
@@ -60,23 +60,23 @@ public Hours getHoursBetween() {
         return Hours.ONE;
     }
 
-    public Optional<Lesson> getLesson() {
-        return lesson;
+    public Optional<Commitment> getCommitment() {
+        return commitment;
     }
 
     public boolean isFree() {
-        return lesson.isEmpty();
+        return commitment.isEmpty();
     }
 
     /**
      * Sets this slot to hold the current lesson.
-     * @param lesson
+     * @param commitment
      */
-    public void setLesson(Lesson lesson) {
+    public void setCommitment(Commitment commitment) {
         if (!isFree()) {
             throw new CommitmentClashException(ALREADY_FILLED_MESSAGE);
-        } else if (canFitLesson(lesson)) {
-            this.lesson = Optional.ofNullable(lesson);
+        } else if (canFitCommitment(commitment)) {
+            this.commitment = Optional.ofNullable(commitment);
```

Comment:
Alright, thanks for changing to `Commitment`. Now if we want, we can expand upon it (I doubt we have time for it).